### PR TITLE
Simplify and update README.md

### PR DESCRIPTION
**Pull request by Nirusu**
The old README.md were like initial research notes. They might not be 100% accurate (not sure if they were updated) and not very useful to any consumers of the library.

Here is my attempt at simplifying the README.md reflecting the current state of the library.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,100 +11,25 @@ This library comes in two parts:
...
     Verify a quote issued by an SGX/TDX TEE
 
-This repo provides a simple Go library to enable these features.
+This repo provides a simple Go library to enable these features and is used by [Constellation](https://github.com/edgelesssys/constellation) to enable TDX attestation.
 
-## Quote Verification
+## Restrictions
+- The current version is based on [DCAP 1.15](https://github.com/intel/SGXDataCenterAttestationPrimitives/releases/tag/DCAP_1.15). 
+and has been tested on the kernel and libraries from the tdx-tools release [2023ww01](https://github.com/intel/tdx-tools/releases/tag/2023ww01).
+Given that the UAPI for TDX is yet to be finished and upstreamed, newer versions might not work yet.
```
Comment (daniel-weisse) [inline README.md:19]:
```suggestion
- The current version is based on [DCAP 1.15](https://github.com/intel/SGXDataCenterAttestationPrimitives/releases/tag/DCAP_1.15),
and has been tested on the kernel and libraries from the tdx-tools release [2023ww01](https://github.com/intel/tdx-tools/releases/tag/2023ww01).
Given that the UAPI for TDX is yet to be finished and upstreamed, newer versions might not be supported.
```

File: README.md [hunk 2]
```diff
@@ -11,100 +11,25 @@ This library comes in two parts:
...
-To verify a quote, we first have to retrieve a collateral from Intel's Provisioning Certification Service (PCS).
-Using this collateral we can verify a quote.
+- This library only supports a subset of the PCS API v4. SGX and other versions of the API are not supported and currently out of scope.
 
-### Attestation flow
+In case you encounter any issues despite the known restrictions, feel free to open [an issue](https://github.com/edgelesssys/go-tdx-qpl/issues/new/choose).
 
-1. Verify enclave PCK cert chain using PCK CRL chain, Root CA CRL, and trusted Root CA.
+## Examples
+An example quote can be found [here](blobs/quote). Other example and test data can also be found in the [*blobs*](blobs) dictionary, or alternatively directly from Intel [directly](https://github.com/intel/SGXDataCenterAttestationPrimitives/tree/master/QuoteVerification/QVL/Src/AttestationApp/sampleData/tdx).
```
Comment (daniel-weisse) [inline README.md:26]:
```suggestion
## Examples
An example quote can be found [here](blobs/quote). Other example and test data can also be found in the [`blobs??](blobs) directory, or alternatively directly from [Intel's DCAP repo](https://github.com/intel/SGXDataCenterAttestationPrimitives/tree/master/QuoteVerification/QVL/Src/AttestationApp/sampleData/tdx).
```

File: README.md (hunk 1 above)
Comment (Nirusu) [reply README.md]: I made the first one a separate sentence.

File: README.md [hunk 3]
```diff
@@ -15,15 +15,15 @@ This repo provides a simple Go library to enable these features and is used by [
...
+It has been tested with the kernel and libraries from the tdx-tools release [2023ww01](https://github.com/intel/tdx-tools/releases/tag/2023ww01).
+Given that the UAPI for TDX is yet to be finished and upstreamed, newer versions might not be supported.
 
 - This library only supports a subset of the PCS API v4. SGX and other versions of the API are not supported and currently out of scope.
 
 In case you encounter any issues despite the known restrictions, feel free to open [an issue](https://github.com/edgelesssys/go-tdx-qpl/issues/new/choose).
 
 ## Examples
-An example quote can be found [here](blobs/quote). Other example and test data can also be found in the [*blobs*](blobs) dictionary, or alternatively directly from Intel [directly](https://github.com/intel/SGXDataCenterAttestationPrimitives/tree/master/QuoteVerification/QVL/Src/AttestationApp/sampleData/tdx).
+An example quote can be found [here](blobs/quote). Other example and test data can also be found in the [`blobs`](blobs) dictionary, or alternatively directly from [Intel's DCAP repo](https://github.com/intel/SGXDataCenterAttestationPrimitives/tree/master/QuoteVerification/QVL/Src/AttestationApp/sampleData/tdx).
```
Comment (daniel-weisse) [inline README.md:26]:
```suggestion
An example quote can be found [here](blobs/quote). Other example and test data can also be found in the [`blobs`](blobs) directory, or alternatively directly from [Intel's DCAP repo](https://github.com/intel/SGXDataCenterAttestationPrimitives/tree/master/QuoteVerification/QVL/Src/AttestationApp/sampleData/tdx).
```

**Review by daniel-weisse — APPROVED** [review]

**Pull request merged**